plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java

```diff
@@ -291,7 +293,14 @@ private static String buildSplitPath(String tableLocation, AddFileEntry addActio
     {
         // paths are relative to the table location and are RFC 2396 URIs
         // https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-file-and-remove-file
-        String path = URI.create(addAction.getPath()).getPath();
+        URI uri = URI.create(addAction.getPath());
+        String path = uri.getPath();
+
+        // org.apache.hadoop.fs.azurebfs.AzureBlobFileSystem encodes the path as URL
+        // https://issues.apache.org/jira/browse/HADOOP-18580
+        if (tableLocation.startsWith("abfs://") || tableLocation.startsWith("abfss://")) {
+            path = URLDecoder.decode(path.replace("+", "%2B"), UTF_8);
```

Comment:
Why is the `+` symbol special?

Reply:
https://stackoverflow.com/a/5450212 ?

Reply:
Right. For instance, both `/a space` & `/a+space` would be decoded as `/a space` and Azure library throws a file not found exception for `/a+space`. I will leave a comment if needed.

Reply:
Looking at the https://github.com/trinodb/trino/pull/17038 I now think the code shouldn't be here at all.
We should defer ABFS filesstem hacks until we talk to the filesystem, i.e. do this on the worker when opening the split.
This would also make it impossible/hard to apply the logic inconsistently for relative/absolute paths, as per https://github.com/trinodb/trino/pull/17038#discussion_r1188386453



cc @vinay-kl @findinpath @alexjo2144 @ebyhr